vulnfeeds/cpp/main.go

```diff
@@ -190,31 +190,42 @@ func main() {
 
 		Metrics.CVEsForApplications++
 
-		Logger.Infof("%s", cve.CVE.CVEDataMeta.ID)
-		for _, ref := range refs {
-			// Are any of the reference's tags 'Patch'?
-			for _, tag := range ref.Tags {
-				if tag == "Third Party Advisory" {
-					continue
-				}
-				if utility.IsRepoURL(ref.URL) {
-					Logger.Infof("\t * %s", ref.URL)
-					// CVE entries have one set of references, but can have multiple CPEs
+		// We only need to do this if we didn't get a repo from the CPE Dictionary.
+		if _, ok := ReposForCVE[cve.CVE.CVEDataMeta.ID]; !ok {
+			for _, ref := range refs {
+				for _, tag := range ref.Tags {
+					// Third Party Advisories often reference
```

Comment:
This is starting to get quite nested and complicated. Might be a good time to consider breaking this out into a separate function.